core/block.ts

```diff
@@ -1400,21 +1399,51 @@ export class Block implements IASTNodeLocation, IDeletable {
    * Create a human-readable text representation of this block and any children.
    *
    * @param opt_maxLength Truncate the string to this length.
-   * @param opt_emptyToken The placeholder string used to denote an empty field.
+   * @param opt_emptyToken The placeholder string used to denote an empty input.
    *     If not specified, '?' is used.
    * @returns Text of block.
    */
   toString(opt_maxLength?: number, opt_emptyToken?: string): string {
-    const tokens = [];
-    const emptyFieldPlaceholder = opt_emptyToken || '?';
+    const tokens = this.toTokens(opt_emptyToken);
 
-    // Temporarily set flag to navigate to all fields.
-    const prevNavigateFields = ASTNode.NAVIGATE_ALL_FIELDS;
-    ASTNode.NAVIGATE_ALL_FIELDS = true;
+    // Run through our tokens array and simplify expression to remove
+    // parentheses around single field blocks.
+    // E.g. ['repeat', '(', '10', ')', 'times', 'do', '?']
+    for (let i = 2; i < tokens.length; i++) {
+      if (tokens[i - 2] === '(' && tokens[i] === ')') {
+        tokens[i - 2] = tokens[i - 1];
+        tokens.splice(i - 1, 2);
+      }
+    }
 
-    let node = ASTNode.createBlockNode(this);
-    const rootNode = node;
+    let prev = '';
```

Comment:
Can you add the comment too? 

// Join the text array, removing the spaces around added parentheses.